src/app/loans/loans-view/loan-account-actions/approve-loan/approve-loan.component.html

```diff
@@ -0,0 +1,140 @@
+<div class="container">
+
+  <form [formGroup]="approveLoanForm" (ngSubmit)="submit()">
+
+    <mat-card-content>
+
+      <div fxLayout="column">
+
+        <mat-form-field>
+          <mat-label>Approved On</mat-label>
+          <input matInput [min]="minDate" [matDatepicker]="approvedDatePicker" required formControlName="approvedOnDate">
+          <mat-datepicker-toggle matSuffix [for]="approvedDatePicker"></mat-datepicker-toggle>
+          <mat-datepicker #approvedDatePicker></mat-datepicker>
+          <mat-error *ngIf="approveLoanForm.controls.approvedOnDate.hasError('required')">
+            Approved Date <strong>is required</strong>
+          </mat-error>
+        </mat-form-field>
+
+        <mat-form-field>
+          <mat-label>Expected disbursement on</mat-label>
+          <input matInput [min]="minDate" [matDatepicker]="disbursementDatePicker" required formControlName="expectedDisbursementDate">
+          <mat-datepicker-toggle matSuffix [for]="disbursementDatePicker"></mat-datepicker-toggle>
+          <mat-datepicker #disbursementDatePicker></mat-datepicker>
+          <mat-error *ngIf="approveLoanForm.controls.expectedDisbursementDate.hasError('required')">
+            Disbursement Date <strong>is required</strong>
+          </mat-error>
+        </mat-form-field>
+
+        <mat-form-field>
+          <mat-label>Approved Amount</mat-label>
+          <input matInput formControlName="approvedLoanAmount">
+        </mat-form-field>
+
+        <div *ngIf="disbursementDetails.length !== 0">
+
+          <mat-label> Loan Tranche Details </mat-label>
+
+          <table>
```

Comment:
Can you try using material table with forms? 

[Link](https://stackblitz.com/edit/angular-material-table-with-form?file=app%2Fapp.component.html) for a basic implementation of forms with material table, if this helps.